core/src/sumeragi/main_loop.rs

```diff
@@ -367,22 +363,30 @@ impl Sumeragi {
         self.connect_peers(&self.current_topology);
     }
 
-    fn update_state(&mut self, committed_block: &VersionedCommittedBlock) {
+    fn update_state(
+        &mut self,
+        committed_block: &Arc<VersionedCommittedBlock>,
```

Comment:
We don't want to increment the refcount, but at the same time we don't want to generify with an `AsRef`, to not accept any other smart pointer?

Reply:
It's just clippy complaining that this value could be passed by reference instead of value.
I decide to obey instead of adding ignore because see no harm in such signature.